AppCenter/AppCenterTests/MSIngestionSenderTests.m

```diff
@@ -298,28 +291,24 @@ - (void)testTasksRunningOnSenderResumed {
   __block NSArray<NSURLSessionDataTask *> *tasks;
   __block BOOL testFinished = NO;
   [MSHttpTestUtil stubLongTimeOutResponse];
-  MSLogContainer *container1 = [self createLogContainerWithId:@"1"];
-  MSLogContainer *container2 = [self createLogContainerWithId:@"2"];
+  NSArray<MSLogContainer *> *containers =
+      @[ [self createLogContainerWithId:@"1"], [self createLogContainerWithId:@"2"] ];
 
   // Send logs
-  [self.sut sendAsync:container1
-      completionHandler:^(__attribute__((unused)) NSString *batchId, __attribute__((unused)) NSUInteger statusCode,
-                          __attribute__((unused)) NSData *data, __attribute__((unused)) NSError *error) {
-        @synchronized(tasks) {
-          if (!testFinished) {
-            XCTFail(@"Completion handler shouldn't be called as test will finish before the response timeout.");
+  for (NSUInteger i = 0; i < [containers count]; i++) {
+    [self.sut sendAsync:containers[i]
+        completionHandler:^(__attribute__((unused)) NSString *batchId, __attribute__((unused)) NSUInteger statusCode,
+                            __attribute__((unused)) NSData *data, __attribute__((unused)) NSError *error) {
+          @synchronized(tasks) {
+            if (!testFinished) {
+              XCTFail(@"Completion handler shouldn't be called as test will finish before the response timeout.");
+            }
           }
-        }
-      }];
-  [self.sut sendAsync:container2
-      completionHandler:^(__attribute__((unused)) NSString *batchId, __attribute__((unused)) NSUInteger statusCode,
-                          __attribute__((unused)) NSData *data, __attribute__((unused)) NSError *error) {
-        @synchronized(tasks) {
-          if (!testFinished) {
-            XCTFail(@"Completion handler shouldn't be called as test will finish before the response timeout.");
-          }
-        }
-      }];
+        }];
+  }
+
+  // Make sure all log containers are enqueued before suspending sender.
+  [NSThread sleepForTimeInterval:0.5];
```

Comment:
Kinda ugly but also more readable. The things we have to do to have more reliable unit tests. ;)

Reply:
As mentioned offline, this is not a real fix for the issue. I doubt it might be caused by at least one log that was enqueued right before suspend which wasn't actually processed yet. Based on Apple document, initial state value is NSURLSessionTaskStateSuspended and it will be changed upon task process. I just want to make sure all logs are enqueued before suspending. Will check if the test still fails in the future.